File: app/domain/models/graph.py
from typing import Any, Dict, List

from pydantic import BaseModel, Field


class GraphNode(BaseModel):
    id: str
    label: str
    properties: Dict[str, Any] = Field(default_factory=dict)


class GraphEdge(BaseModel):
    source_id: str
    target_id: str
    relation_type: str
    properties: Dict[str, Any] = Field(default_factory=dict)


class ExtractionResult(BaseModel):
    nodes: List[GraphNode]
    edges: List[GraphEdge]

File: frontend/src/types/graph.ts
export interface GraphNodeProperty {
  [key: string]: unknown;
}

export interface GraphNode {
  id: string;
  label: string;
  properties?: GraphNodeProperty;
}

export interface GraphEdge {
  source_id: string;
  target_id: string;
  relation_type: string;
  properties?: GraphNodeProperty;
}

export interface GraphSearchResponse {
  nodes: GraphNode[];
  edges: GraphEdge[];
  hits?: GraphNode[];
}

export interface VisNode {
  id: string;
  label: string;
  title?: string;
  color?: {
    background: string;
    border: string;
  };
}

export interface VisEdge {
  id?: string;
  from: string;
  to: string;
  label: string;
  title?: string;
}
